src/api.js

```diff
@@ -938,16 +922,13 @@ api.declare({
   route:      '/task/:taskId/schedule',
   name:       'scheduleTask',
   stability:  API.stability.stable,
-  scopes:     [
-    [
-      // Legacy scope
+  scopes:     {AnyOf: [
+    'queue:schedule-task:<task.schedulerId>/<task.taskGroupId>/<taskId>',
+    {AllOf: [ // Legacy scopes
```

Comment:
could this have `{if: 'allowLegacyScopes', then: ...}` to suggest that these are legacy scopes (with `allowLegacyScopes: true` in the API implementation).

Reply:
We can do this no-problem when we already call `req.authorize`, but to add it to functions that don't defer auth, we will need to add an otherwise unnecessary `req.authorize` call. Would you still prefer it?

Reply:
Yes, I think so